### ISSUE TITLE
segfault noticed on a node from devnet

### ISSUE THREAD

**Issue by zaibon** [threefoldtech/0-fs#29]
```
2020-07-14T10:58:23.516Z: main I > mount ready 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x704fd2]

goroutine 35 [running]:
github.com/threefoldtech/0-fs/meta.(*Dir).Name(0x0, 0xc0005f4400, 0x8)
	/home/runner/go/src/github.com/threefoldtech/0-fs/meta/dir.go:31 +0x22
github.com/threefoldtech/0-fs/meta.(*sqlStore).get(0xc000096300, 0xc0004e8030, 0x7, 0x10, 0x937c40, 0x5cb101, 0xc000282040)
	/home/runner/go/src/github.com/threefoldtech/0-fs/meta/store.go:253 +0x1e5
github.com/threefoldtech/0-fs/meta.(*sqlStore).Get(0xc000096300, 0xc0004e8030, 0x7, 0xc000282040, 0x1, 0x1)
	/home/runner/go/src/github.com/threefoldtech/0-fs/meta/store.go:267 +0x3f
github.com/threefoldtech/0-fs/rofs.(*filesystem).GetAttr(0xc00000e260, 0xc0004e8030, 0x7, 0xc000200198, 0xc0004e8030, 0x7)
	/home/runner/go/src/github.com/threefoldtech/0-fs/rofs/rofs.go:69 +0x159
github.com/threefoldtech/0-fs/vendor/github.com/hanwen/go-fuse/fuse/pathfs.(*readonlyFileSystem).GetAttr(0xc000072220, 0xc0004e8030, 0x7, 0xc000200198, 0x7, 0x4594ce)
	/home/runner/go/src/github.com/threefoldtech/0-fs/vendor/github.com/hanwen/go-fuse/fuse/pathfs/readonlyfs.go:26 +0x51
github.com/threefoldtech/0-fs/vendor/github.com/hanwen/go-fuse/fuse/pathfs.(*pathInode).Lookup(0xc0004980c0, 0xc000200118, 0xc0004e8010, 0x3, 0xc000200198, 0xc0003101e0, 0x20)
	/home/runner/go/src/github.com/threefoldtech/0-fs/vendor/github.com/hanwen/go-fuse/fuse/pathfs/pathfs.go:546 +0xd2
github.com/threefoldtech/0-fs/vendor/github.com/hanwen/go-fuse/fuse/nodefs.(*FileSystemConnector).internalLookup(0xc000096600, 0xc000200118, 0xc00025e690, 0xc0004e8010, 0x3, 0xc000200180, 0xc000060700, 0xc000282010)
	/home/runner/go/src/github.com/threefoldtech/0-fs/vendor/github.com/hanwen/go-fuse/fuse/nodefs/fsops.go:91 +0xb6
github.com/threefoldtech/0-fs/vendor/github.com/hanwen/go-fuse/fuse/nodefs.(*rawBridge).Lookup(0xc000096600, 0xc000200180, 0xc0004e8010, 0x3, 0xc0002000f0, 0x92d5c0)
	/home/runner/go/src/github.com/threefoldtech/0-fs/vendor/github.com/hanwen/go-fuse/fuse/nodefs/fsops.go:104 +0x93
github.com/threefoldtech/0-fs/vendor/github.com/hanwen/go-fuse/fuse.doLookup(0xc0000b4000, 0xc000200000)
	/home/runner/go/src/github.com/threefoldtech/0-fs/vendor/github.com/hanwen/go-fuse/fuse/opcode.go:291 +0x89
github.com/threefoldtech/0-fs/vendor/github.com/hanwen/go-fuse/fuse.(*Server).handleRequest(0xc0000b4000, 0xc000200000, 0xc000200000)
	/home/runner/go/src/github.com/threefoldtech/0-fs/vendor/github.com/hanwen/go-fuse/fuse/server.go:404 +0x30c
github.com/threefoldtech/0-fs/vendor/github.com/hanwen/go-fuse/fuse.(*Server).loop(0xc0000b4000, 0xc00053e001)
	/home/runner/go/src/github.com/threefoldtech/0-fs/vendor/github.com/hanwen/go-fuse/fuse/server.go:376 +0x148
created by github.com/threefoldtech/0-fs/vendor/github.com/hanwen/go-fuse/fuse.(*Server).readRequest
	/home/runner/go/src/github.com/threefoldtech/0-fs/vendor/github.com/hanwen/go-fuse/fuse/server.go:284 +0x332

```
It seems to be running this flist when it crashed: https://hub.grid.tf/zaibon/zaibon-ubuntu-ssh-0.0.5.flist
```
g8ufs --version
0-fs version 'development' @Revision: 5f72ccfed2bd6488df456d026cc822a21da11de7
```

**Comment by muhamadazmy** [comment]
- It's so weird that this happens now, a nil pointer exception can't go undetected for so long
- I reviewed the code and didn't get how that is even possible.
- I tried to reproduce this issue using the same flist with no luck.
  - Scanning the file trees
  - Stat each file in the tree
  - Accessing files that does not exist
  - Accessing dirs that does not exist.

It might help us if we know what exactly was running on that flist
